Job validation

nspc97 opened the pull request:
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -275,9 +275,11 @@ public function getActivityDetails(Activity $activity): Response
 
-        if ($rights === false && !$hasAccess) {
+        if (($rights === false && !$hasAccess) ||
+            ($activity->getStatus() === Activity::STATUS_IN_VALIDATION &&
```
A PM has always access to the jobs posted by users for whom they are a PM. Also here you don't check that the current user is a PM of the owner. Also you have remove admin from always having access :)

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -470,6 +474,24 @@ public function createAction(
+            $owner->getName() . ' ' . $owner->getSurname() .
+            ' has created a new activity waiting for your validation '
+        ))
+            ->setFrom($_ENV['EMAIL_FROM'])
```
Inject the parameter `EMAIL_FROM` and make this change everywhere.

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -470,6 +474,24 @@ public function createAction(
+            ->setTo($owner->getProjectManager()->getEmail())
+            ->setBody(
+                $this->renderView(
+                    'mail/mail_for_PM.html.twig',
```
`mail_for_PM` doesn't indicate what mail is this. Is it that a user has been assigned to the PM, is it a new job to validate e.t.c.

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -470,6 +474,24 @@ public function createAction(
 
         $activityRepository->save($newActivity);
 
+        $message = (new Swift_Message(
```
Create a separate service for sending emails.

SpartakusMd commented on src/Repository/ActivityRepository.php:
```diff
@@ -149,4 +155,23 @@ public function getPaginatedActivities(
+            ->join('activity.owner', 'user')
+            ->where('activity.status = :in_validation')
+            ->setParameter(':in_validation', Activity::STATUS_IN_VALIDATION);
+        $hasAccess = $this->checker->isGranted('ROLE_ADMIN');
```
Everywhere the variable name is `$hasAccess`! Why not use something more explicit like `$isAdmin`.

SpartakusMd commented on src/Transformer/ActivityTransformer.php:
```diff
@@ -164,7 +164,11 @@ public function editTransform(
         $activity->setFinalDeadline($dto->finalDeadline);
-        $activity->setStatus($dto->status);
+
+        if ($activity->getStatus() !== Activity::STATUS_IN_VALIDATION) {
```
👍 Very good :)

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

- Created a separate service for sending emails

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

- Renamed hasAccess to isAdmin

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

- Renamed hasAccess to isAdmin

SpartakusMd commented on src/Service/EmailSender.php:
```diff
@@ -0,0 +1,52 @@
+
+class EmailSender
+{
+    use ControllerTrait;
```
ControllerTrait should be used only in Controllers.

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

- Removed ControllerTrait from EmailSender

Commit:
[BE] Jobs for validation list
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list
